examples/textual_inversion/textual_inversion.py

```diff
@@ -511,10 +526,11 @@ def main():
     for epoch in range(args.num_train_epochs):
         text_encoder.train()
         for step, batch in enumerate(train_dataloader):
-            with accelerator.accumulate(text_encoder):
+            with accelerator.autocast(), accelerator.accumulate(text_encoder):
```

Comment:
What would be necessary in order for this to work without autocast?

There's some concern about its use: https://github.com/huggingface/diffusers/pull/511

Reply:
Speed is about 10% better without autocast and VRAM use decreases to 5500 MB. It does need some additional casts and I don't think it's quite identical functionally since some operations that would be autocasted to fp32 are computed in fp16. Results look still fine though so it might be okay. Smaller VRAM use does enable increasing batch size or disabling gradient checkpointing for even bigger speed up.

Reply:
I checked a bit. The main problem is 
```
accelerator.backward(loss)
```
If that can be done in fp16, it should work without autocast